src/java.desktop/share/classes/javax/swing/text/DefaultCaret.java

```diff
@@ -1031,17 +1046,31 @@ public void setVisible(boolean e) {
      * @see Caret#setBlinkRate
      */
     public void setBlinkRate(int rate) {
+        if (rate < 0) {
+            throw new IllegalArgumentException("Invalid blink rate: " + rate);
+        }
         if (rate != 0) {
-            if (flasher == null) {
-                flasher = new Timer(rate, handler);
+            if (component.isEditable()) {
+                if (flasher == null) {
+                    flasher = new Timer(rate, handler);
+                }
+                flasher.setDelay(rate);
+            } else {
+                savedBlinkRate = rate;
+                isBlinkRateSaved = true;
             }
-            flasher.setDelay(rate);
         } else {
             if (flasher != null) {
                 flasher.stop();
                 flasher.removeActionListener(handler);
                 flasher = null;
             }
+            if (component.isEditable()) {
+                if (isBlinkRateSaved) {
```

Comment:
```suggestion
            if (component.isEditable() && isBlinkRateSaved) {
```
Less indentation and the same behaviour.

Reply:
> Less indentation and the same behaviour.

Agreed. Fixed.